pymilvus/orm/connections.py

```diff
@@ -266,96 +219,74 @@ def connect(self, alias=Config.MILVUS_CONN_ALIAS, user="", password="", db_name=
-            in_addr, parsed_uri = self.__get_full_address(*config)
-            kwargs["address"] = in_addr
-
-            if self.has_connection(alias):
-                if self._alias[alias].get("address") != in_addr:
-                    raise ConnectionConfigException(message=ExceptionsMessage.ConnDiffConf % alias)
-
-            # uri might take extra info
-            if parsed_uri is not None:
-                user = parsed_uri.username if parsed_uri.username is not None else user
-                password = parsed_uri.password if parsed_uri.password is not None else password
-
-                group = parsed_uri.path.split("/")
-                db_name = "default"
-                if len(group) > 1:
-                    db_name = group[1]
-
-            # Set secure=True if username and password are provided
-            if len(user) > 0 and len(password) > 0:
-                kwargs["secure"] = True
-
+        address = kwargs.pop("address", "")
+        uri = kwargs.pop("uri", "")
+        host = kwargs.pop("host", "")
+        port =  kwargs.pop("port", "")
+        user = '' if user is None else str(user)
+        password = '' if password is None else str(password)
+        db_name = '' if db_name is None else str(db_name)
+
+        if set([address, uri, host, port]) != {''}:
+            address, user, password, db_name = self.__parse_info(address, uri, host, port, db_name, user, password)
+            kwargs["address"] = address
+
+        elif alias in self._alias:
+            kwargs = dict(self._alias[alias].items())
+            # If user is passed in, use it, if not, use previous connections user.
+            prev_user = kwargs.pop("user")
+            user = user if user != "" else prev_user
+             # If db_name is passed in, use it, if not, use previous db_name.
+            prev_db_name = kwargs.pop("db_name")
+            db_name = db_name if db_name != "" else prev_db_name
 
-            connect_milvus(**kwargs, user=user, password=password, db_name=db_name)
-            return
-
-        # 2nd Priority, connection configs from env
-        if self._env_uri is not None:
-            addr, parsed_uri = self._env_uri
-            kwargs["address"] = addr
-
-            user = parsed_uri.username if parsed_uri.username is not None else ""
-            password = parsed_uri.password if parsed_uri.password is not None else ""
-            # Set secure=True if uri provided user and password
-            if len(user) > 0 and len(password) > 0:
-                kwargs["secure"] = True
+        # No params, env, and cached configs for the alias
+        else:
+            raise ConnectionConfigException(message=ExceptionsMessage.ConnLackConf % alias)
 
-            connect_milvus(**kwargs, user=user, password=password, db_name=db_name)
-            return
+         # Set secure=True if username and password are provided
+        if len(user) > 0 and len(password) > 0:
+            kwargs["secure"] = True
```

Comment:
Any reason we relate the `tls` (`secure`) with the `user` and `password`? In fact, authentication with user/password can be done without `tls`, see also https://wiki.lfaidata.foundation/display/MIL/MEP+27+--+Support+Basic+Authentication.

Reply:
Please remove this. We don't need to set secure=True for user and password authentication. secure=True is for tls secure connections.